src/server/stream_family.cc

```diff
@@ -2267,11 +2272,14 @@ void XReadImpl(CmdArgList args, std::optional<ReadOpts> opts, ConnectionContext*
   for (auto& [stream, requested_sitem] : opts->stream_ids) {
     if (auto last_id_it = last_ids->find(stream); last_id_it != last_ids->end()) {
       streamID last_id = last_id_it->second;
-
       if (opts->read_group && !requested_sitem.group) {
         // if the group associated with the key is not found,
-        // we will not read entries from the key.
-        continue;
+        // return NoGroupOrKey error.
+        // We are simply mimicking Redis' error message here...
+        // However, we could actually report more precise error message...
+        (*cntx)->SendError(
```

Comment:
Actually, I didn't want to send an error here because, in case of multiple keys given, we may want to receive entries from existing groups (excluding the non-existent group). Wdyt?

Reply:
if we do not send error here, I'm afraid that may lead to different behavior than Redis (as it returns error right away if any key or group doesn't exist)? did I miss anything?

Reply:
i think you are right.. i just verified this behavior with Redis, will make a change.

Reply:
@Abhra303 after trying more example with Redis, I found Redis also returns no group error as long as there is one group that can't be found for xreadgroup. I tested on Redis with the following unit test (modified from your previous test):

```
  resp = Run({"xreadgroup", "group", "group", "alice", "streams", "mystream", "foo", ">", ">"});
  // returns no group error as "group" was not created for mystream.
  EXPECT_THAT(resp, ArgType(RespExpr::ERROR));
```

Reply:
@theyueli,  Abrha is mostly working on control plane stuff. Therefore, I prefer that Dragonfly's PRs  will be approved by core-team members. @kostasrim , @chakaz , @adiholden are good candidates depending on day of week and holidays :)

You can still consult with Abrhadeep about the changes, of course.

Reply:
> @theyueli, Abrha is mostly working on control plane stuff. Therefore, I prefer that Dragonfly's PRs will be approved by core-team members. @kostasrim , @chakaz , @adiholden are good candidates depending on day of week and holidays :)
> 
> You can still consult with Abrhadeep about the changes, of course.

sounds good! just tagged the core team members as reviewers.